mlir/lib/Dialect/Rock/Transforms/GemmToGridwise.cpp

```diff
@@ -223,6 +235,49 @@ GemmRewritePattern::matchAndRewrite(GemmOp op, GemmOpAdaptor adaptor,
   return success();
 }
 
+LogicalResult GemmRewritePattern::computeGridSize(ConversionPatternRewriter &rw,
+                                                  GemmOp op, Value a,
+                                                  Value b) const {
+  GemmFeatures features = op.getGemmFeatures();
+  Attribute params = op.getParams().value();
+
+  auto aShape = a.getType().cast<MemRefType>().getShape();
+  auto bShape = b.getType().cast<MemRefType>().getShape();
+
+  int64_t G = aShape[0];
+  int64_t M = aShape[2];
+  int64_t N = bShape[2];
+
+  auto obtainGridSize = [&](auto tuningParams) {
+    return (M / tuningParams.getMPerBlock()) *
+           (N / tuningParams.getNPerBlock()) * G;
+  };
+
+  int64_t gridSize{0};
+  if (isAccel(features)) {
+    auto tuningParams = params.cast<RockAccelTuningParamAttrInterface>();
+    gridSize = obtainGridSize(tuningParams);
+
+    if (auto bwdOp = llvm::dyn_cast<Conv2DBwdWeightOp>(*op)) {
+      if (!bwdOp.getKBlocks().has_value()) {
+        LLVM_DEBUG(llvm::dbgs() << "must have kBlocks set at lowering.\n");
+        return failure();
+      }
+      int64_t gemmKBlocks = bwdOp.getKBlocks()->getZExtValue();
+      gridSize *= gemmKBlocks;
+    }
+  } else {
+    auto tuningParams = params.cast<GeneralGemmParamsAttr>();
+    gridSize = obtainGridSize(tuningParams);
```

Comment:
This should be already be part of G as per 
https://github.com/ROCm/rocMLIR/blob/fb40071f408a5b1665f6846876c58d0210e8c710/mlir/lib/Dialect/Rock/Transforms/ConvToGemm.cpp#L586

This seems like over counting...

Reply:
as in there is no need for specially accounting for kblocks here.

Reply:
Yes, makes sense. Thanks for noticing.

Done